File: bench/Reference/tak.erl
#!/usr/bin/env escript
%% tak 30 20 10

main(_) ->
    io:format("~p~n", [tak(30, 20, 10)]).

tak(X, Y, Z) when Y < X -> tak(tak(X - 1, Y, Z), tak(Y - 1, Z, X), tak(Z - 1, X, Y));
tak(_, _, Z) -> Z.

File: bench/Reference/tak.java
class tak {
    static long tak(long x, long y, long z) {
        if (y >= x) return z;
        return tak(tak(x-1, y, z), tak(y-1, z, x), tak(z-1, x, y));
    }
    public static void main(String[] args) { System.out.println(tak(30, 20, 10)); }
}
